test/Dialect/ESI/errors.mlir

```diff
@@ -43,3 +43,35 @@ hw.module @test(%m : !sv.modport<@IData::@Noexist>) {
   // expected-error @+1 {{Could not find modport @IData::@Noexist in symbol table.}}
   %idataChanOut = esi.wrap.iface %m: !sv.modport<@IData::@Noexist> -> !esi.channel<i32>
 }
+
+// -----
+
+esi.service.decl @HostComms {
+  esi.service.to_server @Send : !esi.channel<i16>
+  esi.service.to_client @Recv : !esi.channel<i32>
+}
+
+hw.module @Loopback (%clk: i1) -> () {
+  %dataIn = esi.service.req.to_client <@HostComms::@Recv> (["loopback_tohw"]) : !esi.channel<i32>
+  // expected-error @+1 {{Request type does not match port type '!esi.channel<i16>'}}
```

Comment:
Checks for `EmitOpError`s should also match on the op name in the error output, to ensure that the error is attached to the right op. e.g.:
```
// expected-error @+1 {{'esi.service.req.to_server' op Request type does not match port type '!esi.channel<i16>'}}
```